rpc/mux_broker.go

```diff
@@ -95,9 +94,12 @@ func (m *muxBroker) Dial(id uint32) (net.Conn, error) {
 	return stream, nil
 }
 
-// NextId returns a unique ID to use next.
+// NextId returns a unique ID to use next. There is no need for seeding the
+// rand package as the returned ID's aren't stored or used anywhere outside
+// the current runtime. So it's perfectly fine to get the same pseudo-random
+// numbers each time terraform is running.
 func (m *muxBroker) NextId() uint32 {
-	return atomic.AddUint32(&m.nextId, 1)
+	return rand.Uint32()
```

Comment:
Ah, that's a sharp one... I totally didn't think about that! Of course your very right, it will just wrap and so there is no hard limit... Except maybe for [this part](https://github.com/hashicorp/yamux/blob/master/session.go#L154) here in the yamux code. Seems that check would prevent the wrapping to occur right?